gateway-topology-simple/src/main/java/org/apache/knox/gateway/topology/simple/TopologyComparisonUtil.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership. The ASF
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.knox.gateway.topology.simple;
+
+import org.apache.knox.gateway.topology.Application;
+import org.apache.knox.gateway.topology.Provider;
+import org.apache.knox.gateway.topology.Service;
+import org.apache.knox.gateway.topology.Topology;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Utility for comparing Topology objects for equivalence.
+ */
+class TopologyComparisonUtil {
```

Comment:
Is this class necessary? Can't we just ask if the two topology objects are equal? Most of this logic looks like it should be in an topology equals method.

Reply:
Assuming the JSON representation of `topology1` and `topology2` are the same in case they are equal you can simply do a test like:
`org.apache.knox.gateway.util.JsonUtils.renderAsJsonString(topology1).equals(JsonUtils.renderAsJsonString(topology2)`

Reply:
This thought had crossed my mind as I was implementing it; I think the fact that this requirement doesn't care about the URI, default service path or timestamp is what made me opt for this. I think a complete equals() implementation would include at least the URI, but if others disagree, I could be persuaded to move this logic into the Topology equals() method.

Reply:
I'm already converting JSON to a Topology object for this purpose, and would prefer to avoid re-converting them back to JSON for the sake of comparison.

Reply:
Moved to Topology#equals() method.